great_expectations/data_context/data_context.py

```diff
@@ -3348,6 +3348,23 @@ def run_profiler_with_dynamic_arguments(
             include_citation=include_citation,
         )
 
+    @usage_statistics_enabled_method(
+        event_name="data_context.run_profiler_on_data",
+    )
+    def run_profiler_on_data(
+        self,
+        name: Optional[str] = None,
+        ge_cloud_id: Optional[str] = None,
+        batch_request: Optional[Union[dict, BatchRequest, RuntimeBatchRequest]] = None,
```

Comment:
Per the spec, this is `Optional`. Shouldn't this be mandatory if we're "running on data"?

Reply:
@cdkini Great question.  How about we make it "mandatory" (as in "if you really meant to be running the profiler on data, then please provide the proxy for your data!").  Thanks!

Reply:
Done!